packages/rspack-cli/src/rspack-cli.ts

```diff
@@ -83,20 +95,40 @@ export class RspackCLI {
 		await this.program.parseAsync(hideBin(argv));
 	}
 	async registerCommands() {
-		const builtinCommands = [new BuildCommand(), new ServeCommand()];
+		const builtinCommands = [
+			new BuildCommand(),
+			new ServeCommand(),
+			new PreviewCommand()
+		];
 		for (const command of builtinCommands) {
 			command.apply(this);
 		}
 	}
 	async buildConfig(
 		item: RspackOptions | MultiRspackOptions,
-		options: RspackCLIOptions,
+		options: RspackCLIBuildOptions,
 		rspackEnv: RspackEnv
 	): Promise<RspackOptions | MultiRspackOptions> {
 		const internalBuildConfig = async (item: RspackOptions) => {
 			const isEnvProduction = rspackEnv === "production";
 			const isEnvDevelopment = rspackEnv === "development";
-
+			let entry: Record<string, string> = {};
+			if (!item.entry) {
+				if (options.entry) {
+					entry = {
+						main: options.entry.map(x => path.resolve(process.cwd(), x))[0] // Fix me when entry supports array
+					};
+				} else {
+					const defaultEntryBase = path.resolve(process.cwd(), defaultEntry);
+					const defaultEntryPath =
+						findFileWithSupportedExtensions(defaultEntryBase) ||
+						defaultEntryBase + ".js"; // default entry is js
+					entry = {
+						main: defaultEntryPath
+					};
+				}
+				item.entry = entry;
+			}
```

Comment:
I moved the process of entry field to `buildConfig`, because preview command uses `loadConfig` as well, but `RspackCLIPreviewOptions` doesn't have `entry` option(because it doesn't need).